core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -202,6 +202,18 @@ private[chisel3] trait HasId extends InstanceId {
       setRef(Ref(available_name))
     }
 
+  /** Create a name only if a ref provided, a name is suggested, or
+    * [[autoSeed]] exists.
+    */
+  private[chisel3] def forceOptionalName(namespace: Namespace): Unit =
+    if(_ref.isEmpty) {
+      val candidate_name = computeName(None, None)
+      if (!candidate_name.isEmpty) {
+        val available_name = namespace.name(candidate_name.get)
+        setRef(Ref(available_name))
+      }
+    }
```

Comment:
```suggestion
  private[chisel3] def maybeName(namespace: Namespace): Unit =
    if (_ref.isEmpty) {
      val candidate_name = computeName(None, None)
      if (candidate_name.nonEmpty) {
        val available_name = namespace.name(candidate_name.get)
        setRef(Ref(available_name))
      }
    }
```
A couple of nits: `forceOptionalName` doesn't really make sense, so `maybeName` seems more accurate or perhaps `nameIfSuggestedOrSeeded` if we want to get very explicit 🙂

Reply:
i switched to naming everything, so i am going to just not make this function for now.